Arrowgene.Ddon.GameServer/Characters/ClanManager.cs

```diff
@@ -244,16 +244,31 @@ public void UpdateClan(GameClient client, CDataClanUserParam updateParam, DbConn
             }
         }
 
-        public void JoinClan(uint characterId, uint clanId)
+        public void JoinClan(uint characterId, uint clanId, DbConnection? connectionIn = null)
         {
+            Character character;
             var client = Server.ClientLookup.GetClientByCharacterId(characterId);
-            var character = client.Character;
+
+            if (client == null)
+            {
+                character = Server.CharacterManager.SelectCharacter(characterId, false, connectionIn);
+                if (character == null)
+                {
+                    Logger.Error("Tried to add offline character to clan but select character query failed.");
```

Comment:
This should probably throw a ResponseErrorException with an appropriate error code so the user can see an error happening instead of having the client follow along thinking it succeeded